src/transformers/models/xcodec/configuration_xcodec.py

```diff
@@ -143,44 +134,80 @@ def __init__(
+                f"acoustic_model_config must be a dict or DacConfig instance, but got {type(acoustic_model_config)}"
+            )
 
         if semantic_model_config is None:
             self.semantic_model_config = HubertConfig()
         elif isinstance(semantic_model_config, dict):
-            self.semantic_model_config = HubertConfig(**semantic_model_config)
-        elif isinstance(semantic_model_config, HubertConfig):
+            # could be WavLM or Hubert
+            arch = semantic_model_config.get("architectures", None)
+            if arch and arch[0] == "WavLMModel":
+                self.semantic_model_config = WavLMConfig(**semantic_model_config)
+            elif arch and arch[0] == "HubertModel":
+                self.semantic_model_config = HubertConfig(**semantic_model_config)
+            else:
+                # print warning that defaulting to HubertConfig
+                logger.warning(
+                    f"Could not determine semantic model type from config architectures. Got {arch}. Defaulting to HubertConfig."
+                )
+                self.semantic_model_config = HubertConfig(**semantic_model_config)
+        elif isinstance(semantic_model_config, WavLMConfig) or isinstance(semantic_model_config, HubertConfig):
             self.semantic_model_config = semantic_model_config
+        else:
+            raise ValueError(
+                f"semantic_model_config must be a dict, HubertConfig, or WavLMConfig instance, but got {type(semantic_model_config)}"
+            )
 
         if target_bandwidths is None:
             target_bandwidths = [0.5, 1, 1.5, 2, 4]
 
         self.target_bandwidths = target_bandwidths
-        self.audio_channels = audio_channels
         self.sample_rate = sample_rate
-        self.input_channels = input_channels
-        self.encoder_channels = encoder_channels
         self.kernel_size = kernel_size
         self.channel_ratios = channel_ratios
         self.strides = strides
         self.block_dilations = block_dilations
         self.unit_kernel_size = unit_kernel_size
-        self.decoder_channels = decoder_channels
-        self.output_channels = output_channels
         self.codebook_size = codebook_size
-        self.num_quantizers = num_quantizers
-        self.codebook_dim = codebook_dim
         self.initializer_range = initializer_range
-        self.hidden_dim = hidden_dim
-        self.intermediate_dim = intermediate_dim
-        self.output_dim = output_dim
+        if codebook_dim is None:
+            codebook_dim = self.acoustic_model_config.hidden_size + self.semantic_model_config.hidden_size
+        self.codebook_dim = codebook_dim
+        if input_channels is None:
+            input_channels = self.semantic_model_config.hidden_size
+        self.input_channels = input_channels
+        if encoder_channels is None:
+            encoder_channels = self.semantic_model_config.hidden_size
+        self.encoder_channels = encoder_channels
```

Comment:
1. does the main model really need to know these?
2. it is not possible for `encoder_channels` to be != self.semantic_model_config.hidden_size no?

Reply:
I don't think there's any reason for `input_channels`, `encoder_channels`, `decoder_channels` `output_channels` to be different from `self.semantic_model_config.hidden_size`, so I'll remove those!

Reply:
Created new property to conveniently use within modeling

```python
@property
def semantic_hidden_size(self) -> int:
    return self.semantic_model_config.hidden_size
```